core/app/models/spree/product.rb

```diff
@@ -322,6 +322,10 @@ def taxons_for_store(store)
       end
     end
 
+    def full_in_stock?
```

Comment:
`full_in_stock?` is not the best choice for this method name, maybe `any_variant_in_stock_or_backorderable?` would be better suited